myriad-executor/src/main/java/com/ebay/myriad/executor/MyriadExecutor.java

```diff
@@ -1,285 +1,129 @@
-    /**
-     * Default cpus for executor JVM.
-     */
-    public static final double DEFAULT_CPUS = 0.2;
-
-    public static final Gson GSON = new Gson();
-
-    private static final String PROPERTY_FORMAT = "-D%s=%s ";
-
-    private SlaveInfo slaveInfo;
-
-    private Process process;
-
-    public static void main(String[] args) throws Exception {
-        LOGGER.info("Starting MyriadExecutor...");
-        MesosExecutorDriver driver = new MesosExecutorDriver(new MyriadExecutor());
-        System.exit(driver.run() == Status.DRIVER_STOPPED ? 0 : 1);
-    }
-
-    @Override
-    public void registered(ExecutorDriver driver,
-                           ExecutorInfo executorInfo,
-                           FrameworkInfo frameworkInfo,
-                           SlaveInfo slaveInfo) {
-        LOGGER.debug("Registered ", executorInfo, " for framework ", frameworkInfo, " on mesos slave ", slaveInfo);
-        this.slaveInfo = slaveInfo;
-    }
-
-    @Override
-    public void reregistered(ExecutorDriver driver, SlaveInfo slaveInfo) {
-        LOGGER.debug("ReRegistered");
-    }
-
-    @Override
-    public void disconnected(ExecutorDriver driver) {
-        LOGGER.info("Disconnected");
-    }
 
-    @Override
-    public void launchTask(final ExecutorDriver driver, final TaskInfo task) {
-        // "task id beginning with "yarn" is a mesos task for yarn container
-        if (task.getTaskId().getValue().startsWith("yarn")) {
-            TaskStatus status = TaskStatus.newBuilder().setTaskId(task.getTaskId())
-                .setState(TaskState.TASK_RUNNING).build();
+  private static final Logger LOGGER = LoggerFactory.getLogger(MyriadExecutor.class);
+
+  private static final String YARN_CONTAINER_TASK_ID_PREFIX
+     = "yarn_";
+
+  private Set<String> containerIds;
+
+  public MyriadExecutor(Set<String> containerTaskIds) {
+    this.containerIds = containerTaskIds;
+  }
+
+  @Override
+  public void registered(ExecutorDriver driver, ExecutorInfo executorInfo,
+    FrameworkInfo frameworkInfo, SlaveInfo slaveInfo) {
+      LOGGER.debug("Registered ", executorInfo, " for framework ", frameworkInfo, " on mesos slave ", slaveInfo);
+  }
+
+  @Override
+  public void reregistered(ExecutorDriver driver, SlaveInfo slaveInfo) {
+    LOGGER.debug("ReRegistered");
+  }
+
+  @Override
+  public void disconnected(ExecutorDriver driver) {
+    LOGGER.info("Disconnected");
+  }
+
+  @Override
+  public void launchTask(final ExecutorDriver driver, final TaskInfo task) {
+    LOGGER.debug("launchTask received for taskId: " + task.getTaskId());
+    TaskStatus status = TaskStatus.newBuilder()
+      .setTaskId(task.getTaskId())
+      .setState(TaskState.TASK_RUNNING)
+      .build();
+      driver.sendStatusUpdate(status);
+  }
+
+  @Override
+  public void killTask(ExecutorDriver driver, TaskID taskId) {
+    LOGGER.debug("killTask received for taskId: " + taskId.getValue());
+    TaskStatus status;
+
```

Comment:
We may still need to wait till all mesos tasks corresponding to yarn containers are marked killed as we cannot shutdown the NMTask before that. I think the max number of yarn containers running will be bound by the number of cpu cores (probably not more than 32 or 64 currently). I think it might not be a big problem for an NM that is about to be shutdown. Regarding the containers, it looks like the containers get killed the moment the NM dies. This might not be true with work preserving NM restart but we are not currently supporting that.